frontend/pages/projects/[projectId].tsx

```diff
@@ -242,53 +249,55 @@ export default function ProjectPage({
       image={project ? getImageUrl(project.image) : undefined}
     >
       <BrowseContext.Provider value={contextValues}>
-        {project ? (
-          <div className={classes.contentWrapper}>
-            <div className={classes.mainContent}>
-              <ProjectPageRoot
-                project={{
-                  ...project,
-                  team: members,
-                  timeline_posts: posts,
-                  comments: curComments,
-                }}
-                setMessage={setMessage}
-                isUserFollowing={isUserFollowing}
-                setCurComments={setCurComments}
-                followingChangePending={followingChangePending}
-                likingChangePending={likingChangePending}
-                projectAdmin={members?.find((m) => m.permission === ROLE_TYPES.all_type)}
-                isUserLiking={isUserLiking}
-                numberOfLikes={numberOfLikes}
-                numberOfFollowers={numberOfFollowers}
-                handleFollow={handleFollow}
-                handleLike={handleLike}
-                similarProjects={similarProjects}
-                handleHideContent={handleHideContent}
-                showSimilarProjects={showSimilarProjects}
-                requestedToJoinProject={requestedToJoinProject}
-                handleJoinRequest={handleJoinRequest}
-                hubSupporters={hubSupporters}
-                hubPage={hubPage}
-              />
-            </div>
-            <div className={classes.secondaryContent}>
-              {!smallScreenSize && (
-                <ProjectSideBar
+        <ThemeProvider theme={hubThemeData ? transformThemeData(hubThemeData) : theme}>
```

Comment:
We need an API call to fetch theme data in `[projectId].tsx`, as this page also includes custom theme settings. However, the API call should be conditional and only trigger when we are on specific hubs (after deciding which hub types require theme customization).